crates/nu-color-config/src/shape_color.rs

```diff
@@ -3,143 +3,28 @@ use nu_ansi_term::{Color, Style};
-            } else {
-                Style::new().fg(Color::Cyan)
-            }
-        }
-        "flatshape_globpattern" => {
-            if conf.color_config.contains_key("flatshape_globpattern") {
-                let int_color = &conf.color_config["flatshape_globpattern"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Cyan).bold()
-            }
-        }
-        "flatshape_variable" => {
-            if conf.color_config.contains_key("flatshape_variable") {
-                let int_color = &conf.color_config["flatshape_variable"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Purple)
-            }
-        }
-        "flatshape_flag" => {
-            if conf.color_config.contains_key("flatshape_flag") {
-                let int_color = &conf.color_config["flatshape_flag"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().fg(Color::Blue).bold()
-            }
-        }
-        "flatshape_custom" => {
-            if conf.color_config.contains_key("flatshape_custom") {
-                let int_color = &conf.color_config["flatshape_custom"];
-                lookup_ansi_color_style(int_color.to_string())
-            } else {
-                Style::new().bold()
-            }
-        }
-        _ => Style::default(),
+    match conf.color_config.get(shape.as_str()) {
+        Some(int_color) => lookup_ansi_color_style(int_color.to_string()),
+        None => match shape.as_ref() {
+            "flatshape_garbage" => Style::new().fg(Color::White).on(Color::Red).bold(),
+            "flatshape_bool" => Style::new().fg(Color::LightCyan),
+            "flatshape_nothing" => Style::new().fg(Color::LightCyan),
+            "flatshape_int" => Style::new().fg(Color::Purple).bold(),
+            "flatshape_float" => Style::new().fg(Color::Purple).bold(),
+            "flatshape_range" => Style::new().fg(Color::Yellow).bold(),
+            "flatshape_internalcall" => Style::new().fg(Color::Cyan).bold(),
+            "flatshape_external" => Style::new().fg(Color::Cyan),
+            "flatshape_externalarg" => Style::new().fg(Color::Green).bold(),
+            "flatshape_literal" => Style::new().fg(Color::Blue),
+            "flatshape_operator" => Style::new().fg(Color::Yellow),
+            "flatshape_signature" => Style::new().fg(Color::Green).bold(),
+            "flatshape_string" => Style::new().fg(Color::Green),
+            "flatshape_filepath" => Style::new().fg(Color::Cyan),
+            "flatshape_globpattern" => Style::new().fg(Color::Cyan).bold(),
+            "flatshape_variable" => Style::new().fg(Color::Purple),
+            "flatshape_flag" => Style::new().fg(Color::Blue).bold(),
+            "flatshape_custom" => Style::new().bold(),
+            _ => Style::default(),
```

Comment:
should there maybe be `"flatshape_nothing" => Style::default()` just to be complete?

Reply:
Yeap. Missing that one as well